--- lib/crypt-gost-yescrypt.c.orig	2026-09-11 08:43:51.016615151 -0700
+++ lib/crypt-gost-yescrypt.c	2026-09-11 08:44:18.493071700 -0700
@@ -131,7 +131,7 @@
   intbuf->outbuf[1] = 'g';
 
   /* extract yescrypt output from "$y$param$salt$output" */
-  char *hptr = strchr ((const char *) intbuf->retval + 3, '$');
+  char *hptr = strchr ((char *) intbuf->retval + 3, '$');
   if (!hptr)
     {
       errno = EINVAL;
--- lib/crypt-sm3-yescrypt.c.orig	2026-09-11 08:48:13.544977389 -0700
+++ lib/crypt-sm3-yescrypt.c	2026-09-11 08:48:28.277222184 -0700
@@ -136,7 +136,7 @@
   intbuf->outbuf[3] = '3';
 
   /* extract yescrypt output from "$y$param$salt$output" */
-  char *hptr = strchr ((const char *) intbuf->retval + 3, '$');
+  char *hptr = strchr ((char *) intbuf->retval + 3, '$');
   if (!hptr)
     {
       errno = EINVAL;
