--- configure.ac.new	2025-08-10 09:05:54.172239800 -0700
+++ configure.ac	2025-08-10 09:06:26.990362700 -0700
@@ -81,7 +81,7 @@
 
 # This is always checked (library needs this)
 AC_HEADER_STDC
-AC_FUNC_MALLOC
+#AC_FUNC_MALLOC
 AC_CHECK_FUNC([ioctl], [], [FUNC_NOT_FOUND_LIB([ioctl])])
 AC_CHECK_FUNC([asprintf], [], [FUNC_NOT_FOUND_LIB([asprintf])])
 AC_CHECK_FUNC([scandir], [], [FUNC_NOT_FOUND_LIB([scandir])])
