--- ./loginutils/getty.c.orig	2023-01-03 06:14:43.000000000 -0800
+++ ./loginutils/getty.c	2025-03-02 15:15:41.256517228 -0800
@@ -467,6 +467,9 @@
 	usleep(100*1000); /* 0.1 sec */
 	tcflush(STDIN_FILENO, TCIFLUSH);
 
+	G.tty_attrs.c_lflag = 0;
+	set_tty_attrs();
+
 	/* Prompt for and read a login name */
 	do {
 		/* Write issue file and prompt */
--- mailutils/sendmail.c.orig	2021-12-26 08:53:21.000000000 -0800
+++ mailutils/sendmail.c	2022-09-07 16:47:04.178417356 -0700
@@ -225,6 +225,7 @@
 	char *host = sane_address(safe_gethostname());
 	unsigned nheaders = 0;
 	int code;
+	bool fromalready = false;
 	enum {
 		HDR_OTHER = 0,
 		HDR_TOCC,
@@ -437,6 +438,8 @@
 			continue;
 		}
 
+		if (!strncasecmp(s, "From:", 5)) fromalready = true;
+
 		// analyze headers
 		// To: or Cc: headers add recipients
 		check_hdr = (0 == strncasecmp("To:", s, 3));
@@ -507,6 +510,17 @@
 			// in this case just bail out gracefully
 			if (354 != smtp_check("DATA", -1))
 				goto bail;
+
+			// OpenBSD smtpd, gmail and others require From: in the message header.
+			if (!fromalready)
+			{
+			  char *frombuf = xmalloc(256);
+			  memset(frombuf, 0, 256);
+			  snprintf(frombuf, 255, "From: %s", opt_from);
+			  send_r_n(frombuf);
+			  free(frombuf);
+			}
+
 			// dump the headers
 			while (list) {
 				send_r_n((char *) llist_pop(&list));
--- networking/udhcp/common.c.orig	2020-06-26 11:47:44.000000000 -0700
+++ networking/udhcp/common.c	2020-10-02 14:54:36.619432946 -0700
@@ -32,6 +32,7 @@
 	{ OPTION_IP | OPTION_LIST                 , 0x09 }, /* DHCP_LPR_SERVER    */
 	{ OPTION_STRING_HOST          | OPTION_REQ, 0x0c }, /* DHCP_HOST_NAME     */
 	{ OPTION_U16                              , 0x0d }, /* DHCP_BOOT_SIZE     */
+	{ OPTION_STRING                           , 0x0e }, /* DHCP_MERIT_DUMP    */
 	{ OPTION_STRING_HOST          | OPTION_REQ, 0x0f }, /* DHCP_DOMAIN_NAME   */
 	{ OPTION_IP                               , 0x10 }, /* DHCP_SWAP_SERVER   */
 	{ OPTION_STRING                           , 0x11 }, /* DHCP_ROOT_PATH     */
@@ -106,6 +107,7 @@
 	"lprsrv" "\0"           /* DHCP_LPR_SERVER      */
 	"hostname" "\0"         /* DHCP_HOST_NAME       */
 	"bootsize" "\0"         /* DHCP_BOOT_SIZE       */
+	"meritdump" "\0"	/* DHCP_MERIT_DUMP      */
 	"domain" "\0"           /* DHCP_DOMAIN_NAME     */
 	"swapsrv" "\0"          /* DHCP_SWAP_SERVER     */
 	"rootpath" "\0"         /* DHCP_ROOT_PATH       */
