--- a/Makefile	2023-12-28 21:58:36.447939136 -0800
+++ b/Makefile	2023-12-28 22:00:09.240939142 -0800
@@ -332,7 +332,7 @@
 override PACKAGE_NAME=fpc
 override PACKAGE_VERSION=3.2.2
 REQUIREDVERSION=3.0.4
-REQUIREDVERSION2=3.2.0
+REQUIREDVERSION2=3.2.2
 ifndef inOS2
 override FPCDIR:=$(BASEDIR)
 export FPCDIR
--- a/Makefile.fpc	2023-12-28 21:58:43.410939137 -0800
+++ b/Makefile.fpc	2023-12-28 22:00:34.848939143 -0800
@@ -21,7 +21,7 @@
 
 [prerules]
 REQUIREDVERSION=3.0.4
-REQUIREDVERSION2=3.2.0
+REQUIREDVERSION2=3.2.2
 
 
 # make versions < 3.77 (OS2 version) are buggy
--- a/rtl/linux/aarch64/cprt0.as
+++ b/rtl/linux/aarch64/cprt0.as
@@ -41,10 +41,8 @@ _start:
 	                     init, fini, rtld_fini, stack_end) */
 	adrp	x0,:got:PASCALMAIN
 	ldr	x0,[x0,#:got_lo12:PASCALMAIN]
-	adrp	x3,:got:__libc_csu_init
-	ldr	x3,[x3,#:got_lo12:__libc_csu_init]
-	adrp	x4,:got:__libc_csu_fini
-	ldr	x4,[x4,#:got_lo12:__libc_csu_fini]
+	mov	x3, #0		/* Used to be init.  */
+	mov	x4, #0		/* Used to be fini.  */
 	bl	__libc_start_main
 
 	/* This should never happen */
