--- glibc-2.41/include/libc-symbols.h.orig
+++ glibc-2.41/include/libc-symbols.h
@@ -230,11 +230,7 @@
 #ifdef SHARED
 #define static_link_warning(name)
 #else
-#define static_link_warning(name) static_link_warning1(name)
-#define static_link_warning1(name) \
-  link_warning(name, "Using '" #name "' in statically linked applications \
-requires at runtime the shared libraries from the glibc version used \
-for linking")
+#define static_link_warning(name)
 #endif
 
 /* Declare SYMBOL to be TYPE (`function' or `object') of SIZE bytes
