--- arch/arm64/boot/dts/ti/k3-am625-beagleplay.dts.orig	2025-02-05 10:47:52.000000000 -0800
+++ arch/arm64/boot/dts/ti/k3-am625-beagleplay.dts	2025-02-27 08:33:10.010762864 -0800
@@ -35,9 +35,10 @@
 		mmc2 = &sdhci2;
 		rtc0 = &rtc;
 		rtc1 = &wkup_rtc0;
-		serial0 = &main_uart5;
-		serial1 = &main_uart6;
-		serial2 = &main_uart0;
+		serial0 = &main_uart0;
+		serial1 = &main_uart1;
+		serial5 = &main_uart5;
+		serial6 = &main_uart6;
 		usb0 = &usb0;
 		usb1 = &usb1;
 	};
@@ -167,50 +168,34 @@
 		led-0 {
 			bootph-all;
 			gpios = <&main_gpio0 3 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "heartbeat";
-			function = LED_FUNCTION_HEARTBEAT;
 			default-state = "off";
 		};
 
 		led-1 {
 			bootph-all;
 			gpios = <&main_gpio0 4 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "disk-activity";
-			function = LED_FUNCTION_DISK_ACTIVITY;
-			default-state = "keep";
+			default-state = "off";
 		};
 
 		led-2 {
 			bootph-all;
 			gpios = <&main_gpio0 5 GPIO_ACTIVE_HIGH>;
-			function = LED_FUNCTION_CPU;
+			default-state = "off";
 		};
 
 		led-3 {
 			bootph-all;
 			gpios = <&main_gpio0 6 GPIO_ACTIVE_HIGH>;
-			function = LED_FUNCTION_LAN;
+			default-state = "off";
 		};
 
 		led-4 {
 			bootph-all;
 			gpios = <&main_gpio0 9 GPIO_ACTIVE_HIGH>;
-			function = LED_FUNCTION_WLAN;
-		};
-	};
-
-	gpio_keys: gpio-keys {
-		compatible = "gpio-keys";
-		autorepeat;
-		pinctrl-names = "default";
-		pinctrl-0 = <&usr_button_pins_default>;
-
-		usr: button-usr {
-			label = "User Key";
-			linux,code = <BTN_0>;
-			gpios = <&main_gpio0 18 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "heartbeat";
+			function = LED_FUNCTION_HEARTBEAT;
+			default-state = "off";
 		};
-
 	};
 
 	hdmi0: connector-hdmi {
@@ -530,6 +515,80 @@
 			AM62X_IOPAD(0x01b8, PIN_OUTPUT, 2) /* (C13) SPI0_CS1.EHRPWM0_B */
 		>;
 	};
+
+	// Configure all mikroBUS pins as GPIO.  Invoked by mikrobus-gpio.dtbo.
+
+	mikrobus_gpio: mikrobus-gpio {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x01a0, PIN_INPUT, 7) /* (E18) AN   */
+			AM62X_IOPAD(0x01a8, PIN_INPUT, 7) /* (D20) RST  */
+			AM62X_IOPAD(0x01ac, PIN_INPUT, 7) /* (E19) CS   */
+			AM62X_IOPAD(0x01b0, PIN_INPUT, 7) /* (A20) SCK  */
+			AM62X_IOPAD(0x0194, PIN_INPUT, 7) /* (B19) MISO */
+			AM62X_IOPAD(0x0198, PIN_INPUT, 7) /* (A19) MOSI */
+			AM62X_IOPAD(0x01a4, PIN_INPUT, 7) /* (B20) PWM  */
+			AM62X_IOPAD(0x019c, PIN_INPUT, 7) /* (B18) INT  */
+			AM62X_IOPAD(0x01d8, PIN_INPUT, 7) /* (C15) RX   */
+			AM62X_IOPAD(0x01dc, PIN_INPUT, 7) /* (E15) TX   */
+			AM62X_IOPAD(0x01d0, PIN_INPUT, 7) /* (A15) SCL  */
+			AM62X_IOPAD(0x01d4, PIN_INPUT, 7) /* (B15) SDA  */
+		>;
+	};
+
+	// Configure mikroBUS I2C pins as GPIO.  Invoked by mikrobus-i2c-gpio.dtbo.
+
+	mikrobus_i2c_gpio_pins: mikrobus-i2c-gpio-pins {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x01d0, PIN_INPUT, 7) /* (A15) SCL  */
+			AM62X_IOPAD(0x01d4, PIN_INPUT, 7) /* (B15) SDA  */
+		>;
+	};
+
+	// Configure mikroBUS PWM pins as GPIO.  Invoked by mikrobus-pwm-gpio.dtbo.
+
+	mikrobus_pwm_gpio_pins: mikrobus-pwm-gpio-pins {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x01a4, PIN_INPUT, 7) /* (B20) PWM  */
+		>;
+	};
+
+        // Configure mikroBUS SPI pins as GPIO.  Invoked by mikrobus-spi-gpio.dtbo.
+
+	mikrobus_spi_gpio_pins: mikrobus-spi-gpio-pins {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x01ac, PIN_INPUT, 7) /* (E19) CS   */
+			AM62X_IOPAD(0x01b0, PIN_INPUT, 7) /* (A20) SCK  */
+			AM62X_IOPAD(0x0194, PIN_INPUT, 7) /* (B19) MISO */
+			AM62X_IOPAD(0x0198, PIN_INPUT, 7) /* (A19) MOSI */
+		>;
+	};
+
+	// Configure mikroBUS UART pins as GPIO.  Invoked by mikrobus-uart-gpio.dtbo.
+
+	mikrobus_uart_gpio_pins: mikrobus-uart-gpio-pins {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x01d8, PIN_INPUT, 7) /* (C15) RX   */
+			AM62X_IOPAD(0x01dc, PIN_INPUT, 7) /* (E15) TX   */
+		>;
+	};
+
+	// Configure Grove socket as GPIO.  Invoked by grove-gpio.dtbo.
+
+	grove_gpio_pins: grove-gpio-pins {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x01e8, PIN_INPUT, 7) /* (B17) D0 */
+			AM62X_IOPAD(0x01ec, PIN_INPUT, 7) /* (A17) D1 */
+		>;
+	};
+
+	// Configure Grove socket as UART.  Invoked by grove-serial/.dtbo.
+
+	grove_uart_pins: grove-uart-pins {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x01e8, PIN_INPUT, 1) /* (B17) RXD */
+			AM62X_IOPAD(0x01ec, PIN_OUTPUT, 1) /* (A17) TXD */
+		>;
+	};
 };
 
 &mcu_pmx0 {
@@ -857,6 +916,12 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&mikrobus_spi_pins_default>;
 	status = "okay";
+
+	spidev@0 {
+		compatible = "rohm,dh2228fv";
+		status = "okay";
+		reg = <0>;
+	};
 };
 
 &sdhci0 {
@@ -914,8 +979,9 @@
 };
 
 &main_uart1 {
-	/* Main UART1 is used by TIFS firmware */
-	status = "reserved";
+	pinctrl-names = "default";
+	pinctrl-0 = <&grove_uart_pins>;
+	status = "disabled";
 };
 
 &main_uart5 {
--- drivers/usb/gadget/legacy/hid.c.orig	2025-02-05 10:47:52.000000000 -0800
+++ drivers/usb/gadget/legacy/hid.c	2025-02-25 14:23:09.667847817 -0800
@@ -267,6 +267,48 @@
 	},
 };
 
+/* Additional goop to define a USB raw HID gadget */
+
+static struct hidg_func_descriptor raw_hid_data = {
+        .subclass               = 0, /* No subclass */
+        .protocol               = 0, /* No protocol */
+        .report_length          = 64,
+        .report_desc_length     = 33,
+        .report_desc            = {
+0x06, 0x00, 0xFF, // Usage Page = 0xFF00 (Vendor Defined Page 1)
+0x09, 0x01,       // Usage (Vendor Usage 1)
+0xA1, 0x01,       // Collection (Application)
+// Input report
+0x19, 0x01,       // Usage Minimum
+0x29, 0x40,       // Usage Maximum
+0x15, 0x00,       // Logical Minimum
+0x26, 0xFF, 0x00, // Logical Maximum
+0x75, 0x08,       // Report Size: 8-bit field size
+0x95, 64,         // Report Count
+0x81, 0x02,       // Input (Data, Array, Abs)
+// Output report
+0x19, 0x01,       // Usage Minimum
+0x29, 0x40,       // Usage Maximum
+0x75, 0x08,       // Report Size: 8-bit field size
+0x95, 64,         // Report Count
+0x91, 0x02,       // Output (Data, Array, Abs)
+0xC0              // End Collection
+        }
+};
+
+static struct platform_device raw_hid = {
+        .name                   = "hidg",
+        .id                     = 0,
+        .num_resources          = 0,
+        .resource               = 0,
+        .dev.platform_data      = &raw_hid_data,
+};
+
+int __init init_raw_hid(void)
+{
+	pr_info("g_hid: loading raw HID descriptor\n");
+	return platform_device_register(&raw_hid);
+}
 
 MODULE_DESCRIPTION(DRIVER_DESC);
 MODULE_AUTHOR("Fabien Chouteau, Peter Korsgaard");
@@ -276,6 +318,8 @@
 {
 	int status;
 
+	init_raw_hid();
+
 	status = platform_driver_probe(&hidg_plat_driver,
 				hidg_plat_driver_probe);
 	if (status < 0)
