--- arch/arm/boot/dts/overlays/overlay_map.dts.orig	2024-12-17 01:37:23.000000000 -0800
+++ arch/arm/boot/dts/overlays/overlay_map.dts	2024-12-17 21:38:16.017833560 -0800
@@ -38,6 +38,14 @@
 		bcm2711;
 	};
 
+	disable-ethernet {
+		bcm2711 = "disable-ethernet-pi4";
+	};
+
+	disable-ethernet-pi4 {
+		bcm2711;
+	};
+
 	disable-wifi {
 		bcm2835;
 		bcm2711;
@@ -214,6 +222,14 @@
 		bcm2712;
 	};
 
+	Pi4ClickShield {
+		bcm2711 = "Pi4ClickShield-pi4";
+	};
+
+	Pi4ClickShield-pi4 {
+		bcm2711;
+	};
+
 	pi3-act-led {
 		renamed = "act-led";
 	};
