--- arch/arm/boot/dts/overlays/overlay_map.dts.orig	2024-01-20 20:41:23.975798406 -0800
+++ arch/arm/boot/dts/overlays/overlay_map.dts	2024-01-20 20:45:03.158771680 -0800
@@ -38,6 +38,14 @@
 		bcm2711;
 	};
 
+	disable-ethernet {
+		bcm2711 = "disable-ethernet-pi4";
+	};
+
+	disable-ethernet-pi4 {
+		bcm2711;
+	};
+
 	disable-wifi {
 		bcm2835;
 		bcm2711;
--- drivers/clk/bcm/clk-raspberrypi.c.orig	2023-12-23 09:43:27.536046553 -0800
+++ drivers/clk/bcm/clk-raspberrypi.c	2023-12-23 09:44:47.945092107 -0800
@@ -373,7 +373,7 @@
 		struct raspberrypi_clk_variant *variant;
 
 		if (clks->id >= RPI_FIRMWARE_NUM_CLK_ID) {
-			dev_err(rpi->dev, "Unknown clock id: %u (max: %u)\n",
+			dev_warn(rpi->dev, "Unknown clock id: %u (max: %u)\n",
 					   clks->id, RPI_FIRMWARE_NUM_CLK_ID - 1);
 			return -EINVAL;
 		}
--- drivers/pci/controller/pcie-brcmstb.c.orig	2024-06-12 09:29:38.000000000 -0700
+++ drivers/pci/controller/pcie-brcmstb.c	2024-06-15 08:19:03.204199515 -0700
@@ -1443,7 +1443,7 @@
 		msleep(5);
 
 	if (!brcm_pcie_link_up(pcie)) {
-		dev_err(dev, "link down\n");
+		dev_warn(dev, "link down\n");
 		return -ENODEV;
 	}
 
--- fs/super.c.orig	2024-11-21 06:14:35.000000000 -0800
+++ fs/super.c	2024-11-24 21:07:04.617096129 -0800
@@ -1496,7 +1496,7 @@
 	bdev_handle = bdev_open_by_dev(sb->s_dev, mode, sb, &fs_holder_ops);
 	if (IS_ERR(bdev_handle)) {
 		if (fc)
-			errorf(fc, "%s: Can't open blockdev", fc->source);
+			warnf(fc, "%s: Can't open blockdev", fc->source);
 		return PTR_ERR(bdev_handle);
 	}
 	bdev = bdev_handle->bdev;
@@ -1561,7 +1561,7 @@
 
 	error = lookup_bdev(fc->source, &dev);
 	if (error) {
-		errorf(fc, "%s: Can't lookup blockdev", fc->source);
+		warnf(fc, "%s: Can't lookup blockdev", fc->source);
 		return error;
 	}
 
