--- arch/arm/boot/dts/overlays/overlay_map.dts.orig	2024-12-17 01:37:23.000000000 -0800
+++ arch/arm/boot/dts/overlays/overlay_map.dts	2024-12-17 20:46:41.436384040 -0800
@@ -38,6 +38,14 @@
 		bcm2711;
 	};
 
+	disable-ethernet {
+		bcm2712 = "disable-ethernet-pi5";
+	};
+
+	disable-ethernet-pi5 {
+		bcm2712;
+	};
+
 	disable-wifi {
 		bcm2835;
 		bcm2711;
@@ -214,6 +222,22 @@
 		bcm2712;
 	};
 
+	Pi3ClickShield {
+		bcm2712 = "Pi3ClickShield-pi5";
+	};
+
+	Pi3ClickShield-pi5 {
+		bcm2712;
+	};
+
+	Pi4ClickShield {
+		bcm2712 = "Pi4ClickShield-pi5";
+	};
+
+	Pi4ClickShield-pi5 {
+		bcm2712;
+	};
+
 	pi3-act-led {
 		renamed = "act-led";
 	};
@@ -248,6 +272,14 @@
 		bcm2711;
 	};
 
+	pwmchip0 {
+		bcm2712 = "pwmchip0-pi5";
+	};
+
+	pwmchip0-pi5 {
+		bcm2712;
+	};
+
 	ramoops {
 		bcm2835;
 		bcm2711 = "ramoops-pi4";
--- drivers/clk/bcm/clk-raspberrypi.c.orig	2023-12-23 09:43:27.536046553 -0800
+++ drivers/clk/bcm/clk-raspberrypi.c	2023-12-23 09:44:47.945092107 -0800
@@ -373,7 +373,7 @@
 		struct raspberrypi_clk_variant *variant;
 
 		if (clks->id >= RPI_FIRMWARE_NUM_CLK_ID) {
-			dev_err(rpi->dev, "Unknown clock id: %u (max: %u)\n",
+			dev_warn(rpi->dev, "Unknown clock id: %u (max: %u)\n",
 					   clks->id, RPI_FIRMWARE_NUM_CLK_ID - 1);
 			return -EINVAL;
 		}
--- fs/super.c.orig	2024-11-21 06:14:35.000000000 -0800
+++ fs/super.c	2024-11-24 21:07:04.617096129 -0800
@@ -1496,7 +1496,7 @@
 	bdev_handle = bdev_open_by_dev(sb->s_dev, mode, sb, &fs_holder_ops);
 	if (IS_ERR(bdev_handle)) {
 		if (fc)
-			errorf(fc, "%s: Can't open blockdev", fc->source);
+			warnf(fc, "%s: Can't open blockdev", fc->source);
 		return PTR_ERR(bdev_handle);
 	}
 	bdev = bdev_handle->bdev;
@@ -1561,7 +1561,7 @@
 
 	error = lookup_bdev(fc->source, &dev);
 	if (error) {
-		errorf(fc, "%s: Can't lookup blockdev", fc->source);
+		warnf(fc, "%s: Can't lookup blockdev", fc->source);
 		return error;
 	}
 
