--- dma.h.orig	2021-02-27 19:46:41.499853719 -0800
+++ dma.h	2021-02-27 19:46:53.192040717 -0800
@@ -199,7 +199,6 @@
 void hmac_md5(unsigned char *, int, unsigned char *, int, unsigned char *);
 int smtp_auth_md5(int, char *, char *);
 int smtp_init_crypto(int, int, struct smtp_features*);
-int verify_server_fingerprint(const X509 *);
 
 /* dns.c */
 int dns_get_mx_list(const char *, int, struct mx_hostentry **, int);
