--- abe/config/gcc.conf.orig	2019-06-24 15:16:00.502646701 +0200
+++ abe/config/gcc.conf	2019-06-24 15:16:51.874807654 +0200
@@ -101,7 +101,7 @@
 		    default_configure_flags="${default_configure_flags} --with-fpu=neon-fp-armv8"
 		    ;;
 		arm*-*linux-gnueabihf)
-		    default_configure_flags="${default_configure_flags} --with-fpu=vfpv3-d16"
+		    default_configure_flags="${default_configure_flags} --with-fpu=neon-vfpv3"
 		    ;;
 	    esac
 	fi
@@ -129,10 +129,7 @@
 	;;
     arm*-*linux-gnueabi*)
 	if echo "${gcc_override_configure}" | egrep -q -v 'with-cpu|with-tune' ; then
-	    default_configure_flags="${default_configure_flags} --with-tune=cortex-a9"
-	fi
-	if echo "${gcc_override_configure}" | egrep -q -v 'with-arch|with-cpu' ; then
-	    default_configure_flags="${default_configure_flags} --with-arch=armv7-a"
+	    default_configure_flags="${default_configure_flags} --with-cpu=cortex-a8"
 	fi
 	;;
     aarch64*)
