--- abe/config/gcc.conf.orig	2019-06-24 15:20:34.115503953 +0200
+++ abe/config/gcc.conf	2019-06-24 15:20:48.483548967 +0200
@@ -78,7 +78,7 @@
 esac
 
 # Default languages
-languages="c,c++,fortran,lto"
+languages="ada,c,c++,fortran,lto"
 
 default_makeflags="MAKEINFOFLAGS=--force"
 
@@ -226,7 +223,7 @@
     if test x"${with_languages}" != x"default"; then
 	languages="${with_languages}"
     else
-	languages="c,c++,go,objc,fortran"
+	languages="ada,c,c++"
     fi
 fi	 
 
