--- abe/lib/globals.sh.orig	2020-04-22 18:09:18.149291853 -0700
+++ abe/lib/globals.sh	2020-04-22 18:09:18.157291826 -0700
@@ -260,12 +260,12 @@
 	if is_host_mingw; then
 	    # As Mingw32 requires a cross compiler to be already built, so we
 	    # don't need to rebuild the sysroot.
-            builds="${builds} expat python libiconv binutils libc stage2 gdb qemu"
+            builds="${builds} expat python libiconv binutils libc stage2 gdb"
 	else
 	    # Non-mingw builds skip expat, python and libiconv, but
 	    # are here so that they are included in the manifest, so
 	    # linux and mingw manifests can be identical.
-	    builds="${builds} expat python libiconv binutils stage1 libc stage2 gdb qemu"
+	    builds="${builds} expat python libiconv binutils stage1 libc stage2 gdb"
 	fi
 	if test "$(echo ${target} | grep -c -- -linux-)" -eq 1; then
 	    builds="${builds} gdbserver"
@@ -276,7 +276,7 @@
 	    builds="$(echo ${builds} | sed -e 's: linux::')"
 	fi
     else
-        builds="${builds} binutils stage2 libc gdb qemu" # native build
+        builds="${builds} binutils stage2 libc gdb" # native build
     fi
 
     # if this build is based on a manifest, then we must remove components from
