--- include/configs/sunxi-common.h.orig	2024-12-22 20:58:13.117914822 -0800
+++ include/configs/sunxi-common.h	2024-12-22 21:02:04.149037250 -0800
@@ -287,16 +287,7 @@
 #endif
 
 #define CFG_EXTRA_ENV_SETTINGS \
-	CONSOLE_ENV_SETTINGS \
-	MEM_LAYOUT_ENV_SETTINGS \
-	MEM_LAYOUT_ENV_EXTRA_SETTINGS \
-	DFU_ALT_INFO_RAM \
-	"fdtfile=" FDTFILE "\0" \
-	"console=ttyS0,115200\0" \
-	"uuid_gpt_esp=" UUID_GPT_ESP "\0" \
-	"uuid_gpt_system=" UUID_GPT_SYSTEM "\0" \
-	"partitions=" PARTS_DEFAULT "\0" \
-	BOOTCMD_SUNXI_COMPAT \
-	BOOTENV
+	"bootcmd=mmc dev 0 && load mmc 0:1 0x40100000 boot.scr && source 0x40100000\0" \
+	"console=ttyS0,115200\0"
 
 #endif /* _SUNXI_COMMON_CONFIG_H */
